### PR TITLE
feat: Support delay comparison

### PR DESCRIPTION
**Pull request by pangdayuan1**
For asynchronous interfaces, it is necessary to add delay comparison function

### CODE REVIEW

File: arex-schedule-web-api/src/main/java/com/arextest/schedule/service/PropertiesService.java [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+   * Get the time of delay comparison. Default 60s
+   * @param appId
+   * @return
+   */
+  public int getCompareDelaySeconds(String appId) {
+    if (CollectionUtils.isEmpty(this.propertiesProviders) || StringUtils.isEmpty(appId)) {
+      return delaySeconds;
+    }
+
+    int compareDelaySeconds = propertiesProviders.get(FIRST_INDEX).getCompareDelaySeconds(appId);
```
Comment (QizhengMo) [inline arex-schedule-web-api/src/main/java/com/arextest/schedule/service/PropertiesService.java:33]: 感觉允许 List<Provider> 但是写死 get 0 不太好

直接只允许单个 Provider 或者在 provider 里面加权重属性是否好点？

File: arex-schedule-web-api/src/main/java/com/arextest/schedule/service/PropertiesService.java [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+    return compareDelaySeconds < 1 ? delaySeconds : compareDelaySeconds;
```
Comment (QizhengMo) [inline arex-schedule-web-api/src/main/java/com/arextest/schedule/service/PropertiesService.java:34]: 为什么不允许设置 0 来关掉某个应用的延迟对比？

File: arex-schedule-web-api/src/main/java/com/arextest/schedule/service/ReplayCaseTransmitServiceImpl.java [hunk 3]
```diff
@@ -192,11 +193,11 @@ public void updateSendResult(ReplayActionCaseItem caseItem, CaseSendStatusType s
...
     if (sendStatusType == CaseSendStatusType.SUCCESS) {
       replayActionCaseItemRepository.updateSendResult(caseItem);
-
       // async compare task
-      AsyncCompareCaseTaskRunnable compareTask = new AsyncCompareCaseTaskRunnable(
-          replayResultComparer, caseItem);
-      compareExecutorService.execute(compareTask);
+      int compareDelaySeconds = propertiesService.getCompareDelaySeconds(
+          caseItem.getParent().getAppId());
+      compareExecutorService.schedule(() -> replayCompareService.compareCaseDistributable(caseItem),
```
Comment (QizhengMo) [inline arex-schedule-web-api/src/main/java/com/arextest/schedule/service/ReplayCaseTransmitServiceImpl.java:199]: 请新建一个 Runnable 来执行任务，匿名函数异步会导致 MDC tag 丢失

File: arex-schedule-web-api/src/main/java/com/arextest/schedule/service/ReplayCaseTransmitServiceImpl.java [hunk 4]
```diff
@@ -192,11 +193,11 @@ public void updateSendResult(ReplayActionCaseItem caseItem, CaseSendStatusType s
...
+          compareDelaySeconds, TimeUnit.SECONDS);
```
Comment (QizhengMo) [inline arex-schedule-web-api/src/main/java/com/arextest/schedule/service/ReplayCaseTransmitServiceImpl.java:200]: 同上，考虑允许 0 来关掉延迟对比，关闭的时候走原来的 Runnable

File: arex-schedule-web-api/src/main/java/com/arextest/schedule/service/AsyncDelayCompareCaseTaskRunnable.java [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+    this.replayCompareService = replayCompareService;
+    this.caseItem = caseItem;
+  }
+
+
+  @Override
+  protected void doWithTracedRunning() {
+    boolean compareSuccess = replayCompareService.compareCaseDistributable(caseItem);
+    if (!compareSuccess) {
+      LOGGER.error("Comparer returned false, retry, case id: {}", caseItem.getId());
```
Comment (mr3) [inline arex-schedule-web-api/src/main/java/com/arextest/schedule/service/AsyncDelayCompareCaseTaskRunnable.java:30]: Suggest log  `Compare case failed, case id: {}`

**Review by mr3 — APPROVED** [review]

File: arex-schedule-web-api/src/main/java/com/arextest/schedule/service/PropertiesService.java [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+    if (CollectionUtils.isEmpty(this.propertiesProviders) || StringUtils.isEmpty(appId)) {
+      return delaySeconds;
+    }
+
+    for (PropertiesProvider provider : propertiesProviders) {
+      return provider.getCompareDelaySeconds(appId);
+    }
+
+    return delaySeconds;
+  }
```
Comment (mr3) [inline arex-schedule-web-api/src/main/java/com/arextest/schedule/service/PropertiesService.java:37]: There should not be multiple providers,  please refer to the following code

```java
  @Bean
  @ConditionalOnMissingBean(ConfigProdiver.class)
  public ConfigProdiver configProvider() {
    return new DefaultConfigProdiverImpl(redisUri);
  }
```

**Review by mr3 — APPROVED** [review]

**Review by QizhengMo — APPROVED** [review]

**Pull request merged**